FSharp build is broken due to new Path.GetFileName() overload

## Steps to Reproduce

1. Build FSharp
2. It fails during build

This was most likely introduced because https://github.com/mono/mono/pull/7867 introduced new Span overloads on Path.GetFileName() from corefx.

## Current Behavior

```
"@/external/bockbuild/builds/fsharp-fsharp-561af8b/src/fsharp/FSharp.Build-proto/FSharp.Build-proto.fsproj" (default target) (1) ->
(CopyAndSubstituteTextFiles target) -> 
  @/external/bockbuild/builds/fsharp-fsharp-561af8b/src/scripts/scriptlib.fsx(64,25): error FS0041: A unique overload for method 'GetFileName' could not be determined based on type information prior to this program point. A type annotation may be needed. Candidates: Path.GetFileName(path: ReadOnlySpan<char>) : ReadOnlySpan<char>, Path.GetFileName(path: string) : string [@/external/bockbuild/builds/fsharp-fsharp-561af8b/src/fsharp/FSharp.Build-proto/FSharp.Build-proto.fsproj]
  @/external/bockbuild/builds/fsharp-fsharp-561af8b/src/FSharpSource.targets(207,5): error MSB3073: The command "mono @/external/bockbuild/builds/fsharp-fsharp-561af8b/packages/FSharp.Compiler.Tools.4.1.27/tools/fsi.exe --exec "@/external/bockbuild/builds/fsharp-fsharp-561af8b/src/scripts/subst.fsx" --in:"@/external/bockbuild/builds/fsharp-fsharp-561af8b/src/fsharp/FSharp.Build/Microsoft.FSharp.Targets" --out:"@/external/bockbuild/builds/fsharp-fsharp-561af8b/src/fsharp/FSharp.Build-proto/../../../Proto/net40/bin/Microsoft.FSharp-proto.targets" --pattern1:"{BuildSuffix}" --replacement1:"-proto" --pattern2:"" --replacement2:"" " exited with code 1. [@/external/bockbuild/builds/fsharp-fsharp-561af8b/src/fsharp/FSharp.Build-proto/FSharp.Build-proto.fsproj]

    0 Warning(s)
    2 Error(s)
```

## Expected Behavior

No build failure.

### On which platforms did you notice this

[ X ] macOS
[ ] Linux
[ ] Windows

**Version Used**: master

This is similar to https://github.com/mono/mono/pull/7826

--- a/packaging/MacSDK/fsharp.py
+++ b/packaging/MacSDK/fsharp.py
@@ -8,7 +8,7 @@ class FsharpPackage(GitHubTarballPackage):
             override_properties={ 'make': 'make' })
 
         self.extra_stage_files = ['lib/mono/xbuild/Microsoft/VisualStudio/v/FSharp/Microsoft.FSharp.Targets']
-        self.sources.extend(['patches/fsharp-portable-pdb.patch', 'patches/fsharp-string-switchName.patch'])
+        self.sources.extend(['patches/fsharp-portable-pdb.patch', 'patches/fsharp-string-switchName.patch', 'patches/fsharp-path-overloads.patch'])
 
     def prep(self):
         Package.prep(self)

--- a/packaging/MacSDK/patches/fsharp-path-overloads.patch
+++ b/packaging/MacSDK/patches/fsharp-path-overloads.patch
@@ -1,0 +1,22 @@
+diff --git a/src/scripts/scriptlib.fsx b/src/scripts/scriptlib.fsx
+index cc797e305..ae8a6d3cc 100644
+--- a/src/scripts/scriptlib.fsx
++++ b/src/scripts/scriptlib.fsx
+@@ -59,12 +59,12 @@ module Scripting =
+ 
+     let (++) a b = Path.Combine(a,b)
+ 
+-    let getBasename a = Path.GetFileNameWithoutExtension a
+-    let getFullPath a = Path.GetFullPath a
+-    let getFilename a = Path.GetFileName a
+-    let getDirectoryName a = Path.GetDirectoryName a
++    let getBasename (path: string) = Path.GetFileNameWithoutExtension path
++    let getFullPath (path: string) = Path.GetFullPath path
++    let getFilename (path: string) = Path.GetFileName path
++    let getDirectoryName (path: string) = Path.GetDirectoryName path
+ 
+-    let copyFile source dir =
++    let copyFile (source: string) dir =
+         let dest = 
+             if not (Directory.Exists dir) then Directory.CreateDirectory dir |>ignore
+             let result = Path.Combine(dir, Path.GetFileName source)

